src/functions.cpp

```diff
@@ -1,10 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <algorithm>
 #include <iostream>
 #include <limits>
 
 void print(std::vector<int> const & numbers) {
-    for (int i = 0; i < numbers.size(); ++i) {
+    for (unsigned int i = 0; i < numbers.size(); ++i) {
```

Comment:
The best options is to use std::size_t here.

Reply:
Yup, size_t is just a universal type for any size. You do not have to worry if it's unsigned or unsigned long or anything else, so please always use size_t in such loops.